cypress/e2e/po/pages/extensions.po.ts

```diff
@@ -0,0 +1,249 @@
+    return this.extensionCard(extensionName).getId(`extension-card-uninstall-btn-${ extensionName }`).click();
+  }
+
+  // extension install modal
+  extensionInstallModal() {
+    return this.self().get('[data-modal="installPluginDialog"]');
+  }
+
+  installModalSelectVersionClick(optionIndex: number): Cypress.Chainable {
+    this.extensionInstallModal().getId('install-ext-modal-select-version').click();
+
+    return this.self().get(`.vs__dropdown-menu .vs__dropdown-option:nth-child(${ optionIndex })`).click();
+  }
+
+  installModalCancelClick(): Cypress.Chainable {
+    return this.extensionInstallModal().getId('install-ext-modal-cancel-btn').click();
+  }
+
+  installModalInstallClick(): Cypress.Chainable {
+    return this.extensionInstallModal().getId('install-ext-modal-install-btn').click();
+  }
+
+  // extension uninstall modal
+  extensionUninstallModal() {
+    return this.self().get('[data-modal="uninstallPluginDialog"]');
+  }
+
+  uninstallModalCancelClick(): Cypress.Chainable {
+    return this.extensionUninstallModal().getId('uninstall-ext-modal-cancel-btn').click();
+  }
+
+  uninstallModaluninstallClick(): Cypress.Chainable {
+    return this.extensionUninstallModal().getId('uninstall-ext-modal-uninstall-btn').click();
+  }
+
+  // extension details
+  extensionDetails() {
+    return this.self().getId('extension-details');
+  }
+
+  extensionDetailsBgClick(): Cypress.Chainable {
+    return this.self().getId('extension-details-bg').click();
+  }
+
+  extensionDetailsTitle(): Cypress.Chainable<string> {
+    return this.extensionDetails().getId('extension-details-title').invoke('text');
+  }
+
+  extensionDetailsCloseClick(): Cypress.Chainable {
+    return this.extensionDetails().getId('extension-details-close').click();
+  }
+
+  // extension tabs
+  extensionTabs() {
+    return this.self().getId('extension-tabs');
+  }
+
+  extensionTabInstalledClick(): Cypress.Chainable {
+    return this.extensionTabs().get('li:nth-child(1) a').click();
+  }
+
+  extensionTabAvailableClick(): Cypress.Chainable {
+    return this.extensionTabs().get('li:nth-child(2) a').click();
+  }
+
+  extensionTabUpdatesClick(): Cypress.Chainable {
+    return this.extensionTabs().get('li:nth-child(3) a').click();
+  }
+
+  extensionTabAllClick(): Cypress.Chainable {
+    return this.extensionTabs().get('li:nth-child(4) a').click();
+  }
+
+  // extension reload banner
+  extensionReloadBanner() {
+    return this.self().getId('extension-reload-banner');
+  }
+
+  extensionReloadClick(): Cypress.Chainable {
+    return this.extensionReloadBanner().getId('extension-reload-banner-reload-btn').click();
+  }
+
+  // extension menu
+  private extensionMenu() {
+    return this.self().getId('extensions-page-menu');
+  }
+
+  extensionMenuToggle(): Cypress.Chainable {
+    return this.extensionMenu().click();
+  }
+
+  manageReposClick(): Cypress.Chainable {
+    return this.self().getId('action-menu-0-item').click();
```

Comment:
`action-menu` stuff can use the `cypress/e2e/po/components/action-menu.po.ts` PO

Reply:
Done

Reply:
For neatness, `new ActionMenuPo(this.self())` could be a property on the class

```
actionMenu: new ActionMenuPo(<string selector>)
```